src/software/ai/hl/stp/tactic/attacker/attacker_tactic.cpp

```diff
@@ -45,8 +44,7 @@ void AttackerTactic::updatePrimitive(const TacticUpdate& tactic_update, bool res
     if (reset_fsm)
     {
         fsm_map[tactic_update.robot.id()] = std::make_unique<FSM<AttackerFSM>>(
-            DribbleFSM(ai_config.dribble_tactic_config()),
-            AttackerFSM(ai_config.attacker_tactic_config()));
+            DribbleFSM(ai_config.dribble_tactic_config()), AttackerFSM(ai_config));
```

Comment:
This is the original code writers section below, but why don't you also need to update the DribbleFSM control params as well?

Reply:
The code here is actually not the ControlParams, rather, it's just creating the FSM objects. In this case we're creating an AttackerFSM and from my understanding of how we've setup boost SML (the FSM library we use), we need to pass in the sub FSMs that the FSM uses + the FSM itself as arguments to the constructor of the `FSM<XXXXFSM>`. 

So in this case, `AttackerFSM` has `DribbleFSM` as a sub-FSM, so to generate `FSM<AttackerFSM>`, we need to pass in both the `DribbleFSM` and `AttackerFSM` objects.

Regarding the change I made, in this PR I've update `AttackerFSM` to take in `TbotsProto::AiConfig` (part of our dynamic parameters), rather than just taking the `TbotsProto::AttackerTacticConfig`. You can have a look at attacker FSM and see that I needed some of the other constants from AI config, so I had to make this change.